lib/maze/client/appium/base_client.rb

```diff
@@ -18,13 +18,23 @@ def start_session
           # Ensure the device is unlocked
           Maze.driver.unlock
 
+          # Write Maze's address to file and push to the device
+          config = Maze::Internal::FixtureConfig.new
+          config.add('maze_address', maze_address)
+          Maze::Api::Appium::FileManager.new.write_app_file(config.to_s,
+                                                            Maze::Internal::FixtureConfig::DEVICE_FILENAME)
+
```

Comment:
Thanks to @yousif-bugsnag for realising that this needs to be executed before every scenario, not just once at the start.  Some new plumbing is needed to achieve that, so that the client is aware of each scenario starting, but it makes sense to have as there are other things we need to run before each scenario.